Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -59,7 +59,12 @@ public RestChannel(RestServiceCollection serviceCollection, string endpoint, Cha
         {
             var restMethod = _serviceCollection.GetRestMethod(method);
 
-            var cancellationTokenSource = new CancellationTokenSource();
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+            if (options.Deadline.HasValue)
+            {
+                var timeout = (int) Math.Floor((options.Deadline.Value - DateTime.Now).TotalMilliseconds);
```

Comment:
This should definitely be DateTime.UtcNow - or better, use an injected `IClock` if possible. (We can leave it at `DateTime.UtcNow` for the moment though.)

Reply:
This one is still in development and I should've converted it to Draft since I am still trying to test it with Compute.